### DISCUSSION TITLE
Provide a user API for retrieving additional parameters from request cookies

### DISCUSSION THREAD

**Discussion by jackm** [encode/httpx#1510] [Ideas]
### Checklist

- [x] There are no similar issues or pull requests for this yet.
- [x] I discussed this idea on the [~~community~~discussions chat](https://github.com/encode/httpx/discussions) and feedback is positive.

### Is your feature related to a problem? Please describe.

Provide a user API for retrieving httpx.Cookie object parameters from request cookies such as cookie expiry date, cookie path, secure flag, etc.
Right now it is only possible to get a cookie's value via `httpx.Cookies.get()`. No other cookie parameters are exposed via a user accessible API.

## Describe alternatives you considered

The only way to get these parameters is to manually pull them out of the cookie jar by accessing the internal variables. This should be discouraged as internal implementation details are not guaranteed to be stable after each new version. Also due to the way objects in the cookie jar are structured you need to know the cookie path ahead of time to use as a dict key in order to access the httpx.Cookie object within.

e.g.
```
>>> r = httpx.get('http://httpbin.org/cookies/set?chocolate=chip', allow_redirects=False)
>>> r.cookies.jar._cookies[r.request.url.host]['/']['chocolate']
Cookie(version=0, name='chocolate', value='chip', port=None, port_specified=False, domain='httpbin.org', domain_specified=False, domain_initial_dot=False, path='/', path_specified=True, secure=False, expires=None, discard=True, comment=None, comment_url=None, rest={}, rfc2109=False)
```

## Additional context

See [discussions/1497](https://github.com/encode/httpx/discussions/1497#discussion-3248954)

**Comment by lovelydinosaur** [comment]
Okay, so this still needs to be in the "discussion" category. I'm not necessarily sold on us treating this as an issue.
I'd like to see some assessment of what other Python HTTP libraries do or don't provide in this area.

Also, it'd be helpful to start from a more clear specific motivation. Give an example use-case that you're trying to resolve that you're currently unable to do.

**Reply by jackm** [reply]
The motivation is that when I use httpx and make an HTTP request which results in setting a new cookie, I want to get the cookie expiry date to check if and when the cookie will expire. I also want to get the cookie path value to I know what path this cookie is meant for. Right now there is no official way to get these values.

**Reply by lovelydinosaur** [reply]
So two things here. Hopefully can try to guide this into more useful territory...

1. To re-iterate: I'd like to see some assessment of what other Python HTTP libraries do or don't provide in this area.
2. I'd need a clearer *real world use case* understanding of what you're trying to do, rather than "I need this API, because I want to use this API".
